Assets/BossRoom/Scripts/Client/AnimationCallbacks/AnimatorTriggeredSpecialFX.cs

```diff
@@ -282,18 +213,12 @@ private void OnValidate()
                     m_EventsOnNodeEntry[i].m_AnimatorNodeNameHash = Animator.StringToHash(m_EventsOnNodeEntry[i].m_AnimatorNodeName);
                 }
             }
-            if (m_EventsOnNodeExit != null)
-            {
-                for (int i = 0; i < m_EventsOnNodeExit.Length; ++i)
-                {
-                    m_EventsOnNodeExit[i].m_AnimatorNodeNameHash = Animator.StringToHash(m_EventsOnNodeExit[i].m_AnimatorNodeName);
-                }
-            }
 
             if (m_AudioSources == null || m_AudioSources.Length == 0) // if we have AudioSources handy, plug them in automatically
+            {
                 m_AudioSources = GetComponents<AudioSource>();
+            }
         }
-#endif
 
     }
 
```

Comment:
I did my own little investigation, and there's a neat way to get around serializing properties within the editor with [SerializedProperty](https://docs.unity3d.com/ScriptReference/SerializedProperty.html).

So my suggestion is to move code within these blocks to a new Editor script that lives within Scripts/Editor/. With some tests, I also had to add BossRoom.Client to the Editor's asmdef. Anyways, I'll follow up with @dwoodruffsf offline with the suggested changes, as I got a little too deep into this and I resulted with a working prototype.

Reply:
That sounds great!

Reply:
This will go in another PR. Seems like an architectural change whereas this PR should be dedicated to visuals.